src/test/java/seedu/address/model/event/RecurringEventListTest.java

```diff
@@ -11,31 +15,51 @@ class RecurringEventListTest {
 
     private class RecurringEventStub extends RecurringEvent {
 
-        public RecurringEventStub(String eventName) {
-            super(eventName);
+        public RecurringEventStub(String eventName, String dayOfWeek, String startTime, String endTime) {
+            super(eventName, dayOfWeek, startTime, endTime);
         }
 
         @Override
         public String toString() {
             return this.getEventName();
         }
 
-        @Override
         public int compareTo(RecurringEvent o) {
-            return this.getEventName().compareTo(o.getEventName());
+            return super.compareTo(o);
         }
+
     }
+
     @Test
     void insert_nullEvent_throwsNullPointerException() {
         assertThrows(NullPointerException.class, () -> recurringEventList.insert(null));
     }
 
     @Test
     void testToString() {
-        recurringEventList.insert(new RecurringEventStub("Biking"));
-        recurringEventList.insert(new RecurringEventStub("Skiing"));
-        recurringEventList.insert(new RecurringEventStub("Canoeing"));
+        recurringEventList.insert(new RecurringEventStub("Biking", "MONDAY", "14:00", "15:00"));
+        recurringEventList.insert(new RecurringEventStub("Skiing", "SATURDAY", "09:00", "12:00"));
+        recurringEventList.insert(new RecurringEventStub("Canoeing", "WEDNESDAY", "12:00", "17:00"));
+        recurringEventList.insert(new RecurringEventStub("Jogging", "MONDAY", "07:00", "08:00"));
+        recurringEventList.insert(new RecurringEventStub("Invalid", "MONDAY", "06:00", "08:00"));
+        assertEquals("Jogging\n" + "Biking\n" + "Canoeing\n" + "Skiing\n", recurringEventList.toString());
+    }
 
-        assertEquals("Biking\n" + "Canoeing\n" + "Skiing\n", recurringEventList.toString());
+    @Test
+    void testListBetweenOcurrence() {
+        recurringEventList.insert(new RecurringEventStub("Biking", "MONDAY", "14:00", "15:00"));
+        recurringEventList.insert(new RecurringEventStub("Skiing", "SATURDAY", "09:00", "12:00"));
+        recurringEventList.insert(new RecurringEventStub("Canoeing", "WEDNESDAY", "12:00", "17:00"));
+        recurringEventList.insert(new RecurringEventStub("Jogging", "MONDAY", "07:00", "08:00"));
+        recurringEventList.insert(new RecurringEventStub("Invalid", "MONDAY", "06:00", "08:00"));
+
+        RecurringEvent x = new RecurringEventStub("Invalid", "TUESDAY", "06:00", "08:00");
+
+        LocalDateTime startPeriod = LocalDateTime.of(LocalDate.parse("2023-03-06"), LocalTime.parse("06:00"));
+        LocalDateTime endPeriod = LocalDateTime.of(LocalDate.parse("2023-03-09"), LocalTime.parse("12:00"));
+
```

Comment:
I like the use of test cases! Great work!

Perhaps you can use some of the test utils made by @gmengqi or make some new ones! This will make testing easier in the future and also make it easy to see what the outcome of a particular test case should be.